File: ddmlib/src/main/java/com/android/ddmlib/input/android/InputDevice.java
package com.android.ddmlib.input.android;

import java.util.List;

public class InputDevice {

    public static final boolean EVENT_RAW_MODE = true;

    private String devFile, name;

    InputDevice(List<String> str) {
        super();
        for (int i = 0; i < str.size(); i++) {
            String s = str.get(i);
            if (s.contains(":")) {
                String[] split = s.split(":");
                String name = split[0].trim();
                if (name.startsWith("input props")) {
                    System.out.println("input props==" + str.get(i + 1));
                }
                if (split.length < 2)
                    continue;
                String value = split[1].trim();
                if (name.startsWith("add device")) {
                    devFile = value;
                } else if (name.startsWith("name")) {
                    this.name = value;
                }
            }
        }
    }


    @Override
    public String toString() {
        return "InputDevice \ndev=" + devFile + ", \nname=" + name;
    }

    public String getDevFile() {
        return devFile;
    }

    public String getName() {
        return name;
    }

}

File: ddmlib/src/main/java/com/android/ddmlib/input/android/InputDevice.kt
package com.android.ddmlib.input.android

class InputDevice internal constructor(str: List<String>) {

    var devFile: String? = null
        private set
    var name: String? = null
        private set

    init {
        for (i in str.indices) {
            val s = str[i]
            if (s.contains(":")) {
                val split = s.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                val name = split[0].trim { it <= ' ' }
                if (name.startsWith("input props")) {
                    println("input props==" + str[i + 1])
                }
                if (split.size < 2)
                    continue
                val value = split[1].trim { it <= ' ' }
                if (name.startsWith("add device")) {
                    devFile = value
                } else if (name.startsWith("name")) {
                    this.name = value
                }
            }
        }
    }


    override fun toString(): String {
        return "InputDevice \ndev=$devFile, \nname=$name"
    }

    companion object {

        val EVENT_RAW_MODE = true
    }

}